numpy/polynomial/_polybase.py

```diff
@@ -810,6 +813,60 @@ class domain in NumPy 1.4 and ``None`` in later versions.
             return cls(coef, domain=domain, window=window)
 
     @classmethod
+    def interp(cls, func, deg, x=None, args=(), window=None):
+        """Approximate a function using polynomial interpolation.
+
+        Return the series instance that interpolates a given function.
+
+        Parameters
+        ----------
+        func : function
+            The function to be approximated. It must be a function of a single
+            variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+            that can be passed in the `args` parameter.
+        deg : int or 1-D array_like
+            Degree(s) of the fitting polynomials. All terms up to and including the
+            `deg`'th term are included in the interpolation.
+        x : array_like, shape (M,), optional
```

Comment:
There is already a `fit` function that does least squares for arbitrary points. For `deg + 1` points it is equivalent to `pinv`, which also uses a singular value decomposition.

What is special about the Chebyshev points is the equal ripple nature of the fit and that you don't need to specify the points in the function call or scale them to the domain as they are already defined in the window. The new function is essentially a convenient  shortcut.